crates/pglt_lexer/src/diagnostics.rs

```diff
@@ -0,0 +1,63 @@
+use pglt_diagnostics::{Diagnostic, MessageAndDescription};
+use text_size::TextRange;
+
+/// A specialized diagnostic for scan errors.
+///
+/// Scan diagnostics are always **fatal errors**.
+#[derive(Clone, Debug, Diagnostic, PartialEq)]
+#[diagnostic(category = "syntax", severity = Fatal)]
```

Comment:
`severity = Fatal` is important here -> its what we use higher up to determine whether the document can be used or not